### PR TITLE
CMake use target-specific compile flags

### PR DESCRIPTION
**Pull request by dguittet**
Set the compile flags and definitions to be specific to each target. I originally mishandled the compile flag setting and these were getting lost-- targets were missing compile flags! I noticed this since some warnings weren't showing up.

### CODE REVIEW

**Comment by dguittet** [comment]
Building correctly on Windows, Linux & Mac. Warnings are showing up now, though I'm not sure if they were showing up prior to this. @brtietz @sjanzou Were these warnings already showing up in your Build output?

image
image

**Comment by brtietz** [comment]
Yes, those show up on develop for me.

**Review by sjanzou — COMMENTED** [review]
@dguittet, warnings appearing for me on Windows...

**Comment by dguittet** [comment]
Thanks @sjanzou !

**Review by sjanzou — COMMENTED** [review]
Fixed in develop branch of ssc commit e844e946df4ad302bdc7a176a325a1adcc840538

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,71 +41,93 @@ Project(sam_simulation_core VERSION 1.0.0)
...
+    if (MAIN_CFLAGS MATCHES "NOTFOUND")
+        set(MAIN_CFLAGS "")
+    endif()
     if(MSVC)
-        add_compile_options(/W3)
-        add_compile_definitions( LK_USE_WXWIDGETS _CRT_SECURE_NO_DEPRECATE=1 )
-        add_compile_definitions(_CRT_NON_CONFORMING_SWPRINTFS=1 _SCL_SECURE_NO_WARNINGS=1)
-        add_compile_definitions(__WXMSW__ _UNICODE NOPCH )
+        set(MAIN_CFLAGS "${MAIN_CFLAGS} /W3 /DLK_USE_WXWIDGETS /D _CRT_SECURE_NO_DEPRECATE=1 /D _CRT_NON_CONFORMING_SWPRINTFS=1")
+        set(MAIN_CFLAGS "${MAIN_CFLAGS} /D _SCL_SECURE_NO_WARNINGS=1 /D__WXMSW__ /D_UNICODE NOPCH")
```
[the remark this answers is not in the document]
Comment (dguittet) [reply CMakeLists.txt:112]: My mistake! Thanks for fixing that

**Pull request merged**